maestrowf/__init__.py

```diff
@@ -53,3 +60,372 @@ def emit(self, record):
 
 __version_info__ = ("1", "1", "9dev0")
 __version__ = '.'.join(__version_info__)
+
+
+# Register status layout types
+class StatusRendererFactory:
+    """Factory for setting up alternate console status rendering formats"""
+    def __init__(self):
+        self._layouts = {}
+
+    def register_layout(self, layout, renderer):
+        self._layouts[layout] = renderer
+
+    def get_renderer(self, layout):
+        renderer = self._layouts.get(layout)
+
+        # Note, need to wrap renderer in try/catch too, or return default val?
+        if not renderer:
+            raise ValueError(layout)
+
+        return renderer()
+
+
+class BaseStatusRenderer:
```

Comment:
It looks like this should be an abstract class and the `layout` and `render` methods should be `abstractmethods` -- see the `SchedulerScriptAdapter` for an example.

Reply:
oops, my bad! definitely what this should be